apps/editing-toolkit/editing-toolkit-plugin/coming-soon/fallback-coming-soon-page.php

```diff
@@ -41,16 +60,14 @@ function get_redirect_to() {
  * @return string The login URL
  */
 function get_login_url() {
-	$redirect_to = get_redirect_to();
+	$redirect_to = get_wpcom_redirect_to();
 
+	// If we're on WPCOM use a WordPress.com login URL.
 	if ( function_exists( 'localized_wpcom_url' ) ) {
 		return localized_wpcom_url( '//wordpress.com/log-in?redirect_to=' . $redirect_to );
 	}
```

Comment:
This looks like it'll work but it's interesting having all this wpcom specific code in ETK. The approach I took with the sidebar was to have the self-hosted logic be the default and then allow it to be overridden with hooks e.g.

```php
function get_login_url() {
   $login_url = site_url() . '/wp-login.php?redirect_to=' . set_url_scheme( original_request_url() );
   return apply_filters( 'some_good_hook_name_get_login_url', $login_url );
}
```

And then put all the wpcom logic in the wpcom code.

Not that we're leaking much sensitive at the moment, just seems like things like `get_wpcom_redirect_to()` don't belong in ETK.

Reply:
> just seems like things like get_wpcom_redirect_to() don't belong in ETK.

I agree with you. Ultimately, I think it's okay for now since it's something we can iterate.

But if you think it's a blocker, I'm happy to spin up a couple of patches.

Reply:
Not a blocker, this technique wasn't introduced by this PR anyway :)